src/Config/Converters/DatasourceHealthOptionsConvertorFactory.cs

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Azure.DataApiBuilder.Config.HealthCheck;
+using Azure.DataApiBuilder.Config.ObjectModel;
+
+namespace Azure.DataApiBuilder.Config.Converters;
+
+internal class DatasourceHealthOptionsConvertorFactory : JsonConverterFactory
+{
+    /// <inheritdoc/>
+    public override bool CanConvert(Type typeToConvert)
+    {
+        return typeToConvert.IsAssignableTo(typeof(DatasourceHealthCheckConfig));
+    }
+
+    /// <inheritdoc/>
+    public override JsonConverter? CreateConverter(Type typeToConvert, JsonSerializerOptions options)
+    {
+        return new HealthCheckOptionsConverter();
+    }
+
+    private class HealthCheckOptionsConverter : JsonConverter<DatasourceHealthCheckConfig>
+    {
+        /// <summary>
+        /// Defines how DAB reads the data-source's health options and defines which values are
+        /// used to instantiate DatasourceHealthCheckConfig.
+        /// </summary>
+        /// <exception cref="JsonException">Thrown when improperly formatted health check options are provided.</exception>
+        public override DatasourceHealthCheckConfig? Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            if (reader.TokenType == JsonTokenType.Null)
+            {
+                return new DatasourceHealthCheckConfig() { Enabled = true, Name = null, ThresholdMs = HealthCheckConstants.DefaultThresholdResponseTimeMs };
+            }
+
+            if (reader.TokenType is JsonTokenType.StartObject)
+            {
+                bool enabled = true;
+
+                // Refer to DatasourceHealthCheckConfig record definition to define default name value.
+                string? name = null;
+
+                // Refer to DatasourceHealthCheckConfig record definition to define default threshold-ms value.
+                int? threshold_ms = null;
+
+                while (reader.Read())
+                {
+                    if (reader.TokenType is JsonTokenType.EndObject)
+                    {
+                        return new DatasourceHealthCheckConfig()
+                        {
+                            Enabled = enabled,
+                            Name = name,
```

Comment:
if the name is not defined, how do we pick up the Default value? do we need a check similar to threshold_ms here?